packages/api/apollo-server/mutations/assignment.ts

```diff
@@ -0,0 +1,40 @@
+import { assignments } from '../mock-data'
+
+export interface IAddAssignment {
+  id: String
+  name: String
+  requiredFiles?: String[]
+  peerReviewDeadline?: String
+  feedbackQuestions?: String[]
+  groupSize?: Number
+  isActive?: Boolean
+}
+
+// add a new assignment
+export const addAssignment = (_: any, args: IAddAssignment) => {
+  console.log(args)
+  return null
+}
+
+export interface IEditAssignment {
+  id: String
+  name?: String
+  requiredFiles?: String[]
+  peerReviewDeadline?: String
+  feedbackQuestions?: String[]
+  groupSize?: Number
+  isActive?: Boolean
```

Comment:
> Activating/Deactivating an entity should be a separate mutation. Either edit or activate/deactivate- don't do them both at same time.

I think they pretty much to the same thing. i.e. find a record and update it.
Is it better to keep one mutation and let frontend to enforce no edit and activate/deactivate at the same time?

Reply:
Backend should be independent from the frontend. Even if our frontend does enforce this, we would still need to deal with it in the backend in case we get requests outside of our frontend.

Reply:
I think we should treat deactivate another form of deletion. In that case, we just follow CRUD.

Reply:
> I think we should treat deactivate another form of deletion. In that case, we just follow CRUD.

Perhaps we can adopt GraphQL convention? Instead of having `CRUD` operations, we will only have `Read` and `Write` operation.

e.g. In Query, we have a getSomething, and in Mutation, we have a mutateSomething. The mutation is responsible for `write', `update` and `delete`

Reply:
It might be an awful idea. Since it's hard to check required input fields for different Mutations.

Reply:
Usually, you shouldn't be able to update a deleted entity since it no longer "exists"- it should just return 404.
So what happens if delete and update happens in the same time? Or what if you do if you reactivate + update a user at same time?